typesafety/test_interfaces/test_failable/test_single_failable.yml

```diff
@@ -0,0 +1,54 @@
+- case: single_failable_inheritance_correct2
+  disable_cache: true
+  main: |
+    from typing import TypeVar
+    from returns.interfaces.failable import SingleFailable2
+    from returns.primitives.hkt import SupportsKind2
+
+    _FirstType = TypeVar('_FirstType')
+    _SecondType = TypeVar('_SecondType')
+
+    class MyClass(
+        SupportsKind2['MyClass', _FirstType, _SecondType],
+        SingleFailable2[_FirstType, _SecondType],
+    ):
+        ...
+
+
+- case: single_failable_inheretance_correct3
+  disable_cache: true
+  main: |
+    from typing import TypeVar
+    from returns.interfaces.failable import SingleFailable3
+    from returns.primitives.hkt import SupportsKind3
+
+    _FirstType = TypeVar('_FirstType')
+    _SecondType = TypeVar('_SecondType')
+    _ThirdType = TypeVar('_ThirdType')
+
+    class MyClass(
+        SupportsKind3['MyClass', _FirstType, _SecondType, _ThirdType],
+        SingleFailable3[_FirstType, _SecondType, _ThirdType],
+    ):
+        ...
+
+
+- case: single_failable_inheritance_missing
+  disable_cache: true
+  main: |
+    from typing import TypeVar
+    from typing_extensions import final
+    from returns.interfaces.failable import SingleFailable2
+    from returns.primitives.hkt import SupportsKind2
+
+    _FirstType = TypeVar('_FirstType')
+    _SecondType = TypeVar('_SecondType')
+
+    @final
+    class MyClass(
+        SupportsKind2['MyClass', _FirstType, _SecondType],
+        SingleFailable2[_FirstType, _SecondType],
+    ):
+        ...
+  out: |
+    main:10: error: Final class main.MyClass has abstract attributes "apply", "bind", "from_iterable", "from_value", "map", "rescue"
```

Comment:
Why don't we have `empty` prop here? 🤔

Reply:
This is an interesting question, I tried to add `metaclass=ABCMeta` or inherit directly from `ABC` but nothing make this fails!
This code is working without any issue:
```python
>>> from typing import TypeVar
>>> from typing_extensions import final
>>> from returns.interfaces.failable import SingleFailable2
>>> from returns.primitives.hkt import SupportsKind2
>>> 
>>> _FirstType = TypeVar('_FirstType')
>>> _SecondType = TypeVar('_SecondType')
>>> @final
... class MyClass(
...     SupportsKind2['MyClass', _FirstType, _SecondType],
...     SingleFailable2[_FirstType, _SecondType],
... ):
...     ...
... 
>>> MyClass()
```

Reply:
I can't figure out why this is not failing

Reply:
Do you have some idea?

Reply:
`ReaderLike2` works correctly: https://github.com/dry-python/returns/blob/master/typesafety/test_interfaces/test_specific/test_reader/test_reader_like2.yml#L70 It has abstract property `.empty`

The only difference is `self` annotation.